plugins/autoslab.cpp

```diff
@@ -0,0 +1,267 @@
+/* Simple plugin to check for ghosts and automatically queue jobs to engrave slabs for them.
+ *
+ * Enhancement idea: Queue up a ConstructSlab job, then link the engrave slab job to it. Avoids need to have slabs in stockpiles
+ *                   Would require argument parsing, specifying materials
+ * Enhancement idea: Automatically place the slab. This seems like a tricky problem but maybe solveable with named zones?
+ *                   Might be made obsolete by people just using buildingplan to pre-place plans for slab?
```

Comment:
I'm not sure about this -- in its current form, buildingplan doesn't differentiate between engraved and unengraved slabs, so it would likely just steal the unengraved slabs before dwarves have a chance to engrave them. We could certainly modify buildingplan to only choose engraved slabs (or make that an option).

Reply:
It might be better to update buildingplan to allow specifying engraved slabs, and let this plugin just handle making the engravings. Better to leave the job of placing buildings to the plugin that already does it well. You've got a better sense of where things belong though so I'll defer to whatever you think is best.

Reply:
That sounds like a reasonable plan. Could you file a feature request for `buildingplan` for that? Once I get `gui/control-panel` written, `buildingplan` is my next target.

Reply:
Created https://github.com/DFHack/dfhack/issues/2754